span-key-extractor/src/main/java/com/expedia/www/haystack/pipes/key/extractor/SpanKeyExtractor.java

```diff
@@ -26,12 +26,10 @@
 
     String name();
 
-    public void configure(Config config);
+    public void configure(Config config); // sets up the extractor with configuration
 
-    public Optional<String> extract(Span span);
+    public List<Record> getRecords(Span span); // returns list of records containing message, key and producer topic mapping
 
-    public String getKey();
-
-    public List<String> getTopics();
+    public List<String> getProducers();
```

Comment:
Why do we need this when we have the mapping with Record?

Reply:
agreed. But producers being mandatory field in the extractor config file, it seemed fair to provide a getter for the same.